Lib/test/test_fstring.py

```diff
@@ -1044,6 +1047,15 @@ def test_errors(self):
                              r"f'{1000:j}'",
                             ])
 
+    def test_filename_in_syntaxerror(self):
+        # see issue 38964
+        with temp_cwd() as cwd:
+            filename = os.path.join(cwd, 't.py')
```

Comment:
My only suggestion would be to use a filename more unlikely than "t.py" as a substring. Who knows, maybe the error message contains "output.py" in a traceback, or something like that.

Reply:
But filename contains the full path, which on my machine is `/Users/lysnikolaou/Repositories/cpython/build/test_python_10314/tempcwd/t.py`, so that becomes close to impossible, right? Maybe `filename` is badly named?

Reply:
Ah, never mind. I somehow read it as `self.assertIn("t.py", stderr.decode('utf-8'))`. It looks good as-is. Sorry for the noise. The PR looks good to me.